docs/lang/articles/basic/type.md

```diff
@@ -302,16 +84,30 @@ def func(a: ti.f32) -> ti.i64:
     ...
 ```
 
-### Type casts
+### Type promotion
 
-All data types are static in the **Taichi scope**. Therefore, casts are needed when you want to assign a certain type of data to another one.
+When Taichi performs binary operations on different types, the result is a
+promoted type. This is known as type promotion. Following the C programming
```

Comment:
This didn't explain what a promoted type is. Please could you provide more details?